### PR TITLE
Replace timeout in test by duration assertion.

### PR DESCRIPTION
**Pull request by Carreau**
There seem to be some problem when using pytest timeout. This might be
due to the fact that we use  a thread to interrupt the tests in the
middle.

Let try to keep the test going, just measure the duration and fail the
tests if they took too long.

We also remove the dependency on Pytest-timeout.

Here I'm using `time.monotonic()` (wall clock), but we can also switch to
`process_time()` (that exclude sleep-time and is thus closer to
cpu-time, it might be better in there is a lot of task switching and we
just want to make sure no computation is happening.

Mitigate #3869

### CODE REVIEW

**Review by sofroniewn — APPROVED** [review]
Amazing, thanks so much @Carreau! This works for me, but I'll let @tlambert03 weigh in before merge!!

**Comment by tlambert03** [comment]
this is fine, but if dask starts acting up, this will likely take a _long_ time to finish.  I'm certainly fine with it (since I'm also fine just removing these tests altogether 😄) and am approving, but the failure mode will be less "fun" than the fetch-counting fixture used elsewhere.

**Review by tlambert03 — APPROVED** [review]

**Comment by sofroniewn** [comment]
> this is fine, but if dask starts acting up, this will likely take a long time to finish. I'm certainly fine with it (since I'm also fine just removing these tests altogether 😄) and am approving, but the failure mode will be less "fun" than the fetch-counting fixture used elsewhere.

Yeah agreed - if we start running into this we can revisit. I will merge this now. thanks!!

**Pull request merged**
